larray/core/group.py

```diff
@@ -911,20 +915,31 @@ def by(self, length, step=None):
         >>> from larray import Axis, X
         >>> age = Axis(range(10), 'age')
         >>> age[[1, 2, 3, 4, 5]].by(2)
-        (age[1, 2], age[3, 4], age[5])
+        (age[1, 2] >> '1:2', age[3, 4] >> '3:4', age[5] >> '5')
         >>> age[1:5].by(2)
-        (age.i[1:3], age.i[3:5], age.i[5:6])
+        (age.i[1:3] >> '1:2', age.i[3:5] >> '3:4', age.i[5:6] >> '5')
         >>> age[1:5].by(2, 4)
-        (age.i[1:3], age.i[5:6])
+        (age.i[1:3] >> '1:2', age.i[5:6] >> '5')
         >>> age[1:5].by(3, 2)
-        (age.i[1:4], age.i[3:6], age.i[5:6])
+        (age.i[1:4] >> '1:3', age.i[3:6] >> '3:5', age.i[5:6] >> '5')
+        >>> age[1:5].by(3, 2, '{start}->{end}')
+        (age.i[1:4] >> '1->3', age.i[3:6] >> '3->5', age.i[5:6] >> '5')
         >>> X.age[[0, 1, 2, 3, 4]].by(2)
-        (X.age[0, 1], X.age[2, 3], X.age[4])
+        (X.age[0, 1] >> '0:1', X.age[2, 3] >> '2:3', X.age[4] >> '4')
         """
+        def make_group(start, length, name_template):
+            g = self[start:start + length]
+            labels = g.eval()
+            if labels[0] != labels[-1]:
```

Comment:
this test is strange. I **suppose** you wanted to test if len(labels) == 1 but that is problematic if g.eval() a single label, or something like that. In that case, please add a comment. Otherwise, revert to a more obvious test.